v1/src/main/java/com/google/cloud/teleport/spanner/ddl/InformationSchemaScanner.java

```diff
@@ -1231,4 +1241,58 @@ private void listSequenceOptionsPostgreSQL(
           .endSequence();
     }
   }
+
+  private boolean isPlaceHolderDescriptor(DescriptorProto descriptor) {
+    MessageOptions messageOptions = descriptor.getOptions();
+    // 14004 is the extension for placeholder descriptor in spanner.
+    return messageOptions.getUnknownFields().hasField(14004);
+  }
+
+  private Set<String> collectBundleTypes(FileDescriptorSet fds) {
+    Set<String> result = new HashSet<>();
+    for (FileDescriptorProto file : fds.getFileList()) {
+      String filePackage = file.hasPackage() ? file.getPackage() + "." : "";
+      for (DescriptorProto descriptor : file.getMessageTypeList()) {
+        if (isPlaceHolderDescriptor(descriptor)) {
+          continue;
+        }
+        String descriptorName = filePackage + descriptor.getName();
+        result.add(descriptorName);
+      }
+      for (EnumDescriptorProto enumDescriptor : file.getEnumTypeList()) {
+        String descriptorName = filePackage + enumDescriptor.getName();
+        result.add(descriptorName);
+      }
+    }
+    return result;
+  }
+
+  private void addProtoBundleAndDescriptor(Ddl.Builder builder) {
+    Statement queryStatement;
+    switch (dialect) {
+      case GOOGLE_STANDARD_SQL:
+        queryStatement = Statement.of("SELECT PROTO_BUNDLE FROM information_schema.schemata AS s");
```

Comment:
JJ is working on Named schema support [currently](https://github.com/GoogleCloudPlatform/DataflowTemplates/pull/1572), and there may be race condition. Does this query change in case of Named schema? 
cc: @jin-jj

Reply:
No, proto_bundle is database level property. It wouldn't change with named schema.